feat: adding stacks integration guide (#757)
blockstack/docs: src/pages/stacks-blockchain/network.md

---
title: Network
description: Guide to the Stacks 2.0 network
---

## Tokens

> Stacks ("STX") are the native tokens on the Stacks 2.0 blockchain. The smallest fraction of a STX is one micro-STX. 1.000.000 micro-STX make one STX.

Reviewer: ```suggestion
Stacks ("STX") are the native tokens on the Stacks 2.0 blockchain. The smallest fraction of a STX is one micro-STX. 1,000,000 micro-STX make one STX.
```

The German in you coming out 🇩🇪

> STX amounts should be stored as integers (8 bytes long), and represent the amount of micro-STX. For display purposes, micro-STX are divided by 1.000.000 (decimal precision of 6).

Reviewer: ```suggestion
STX amounts should be stored as integers (8 bytes long), and represent the amount of micro-STX. For display purposes, micro-STX are divided by 1,000,000 (decimal precision of 6).
```

## Fees

Fees are used to incentivize miners to confirm transactions on the Stacks 2.0 blockchain. The fee is calculated based on the estimate fee rate and the size of the [raw transaction](http://localhost:3000/stacks-blockchain/transactions#serialization) in bytes. The fee rate is a market determined variable. For the [testnet](/stacks-blockchain/testnet), it is set to 1 microstack.

Fee estimates can obtained through the [`GET /v2/fees/transfer`](https://blockstack.github.io/stacks-blockchain-api/#operation/get_fee_transfer) endpoint:

```shell
curl 'https://stacks-node-api-latest.argon.blockstack.xyz/v2/fees/transfer'
```

The API will respond with the fee rate (as integer):

```js
1;
```

> [The Stacks Transactions JS library](https://github.com/blockstack/stacks-transactions-js) supports fee estimation for ...

Reviewer: ```suggestion
[The Stacks Transactions JS library](https://github.com/blockstack/stacks-transactions-js) supports fee estimation for:
```

- token transfers (`estimateTransfer`)
- contract deploys (`estimateContractDeploy`)
- non read-only contract calls (`estimateContractFunctionCall`)

-> For an implementation using a different language than JavaScript, please review [this reference implementation](https://github.com/blockstack/stacks-transactions-js/blob/master/src/builders.ts#L97).

> ## Nonces

Author: yeah, great idea. thanks!

> Every account carries a nonce property that indicates the number of transactions processed for the given account. Nonces are one-time codes and are by 1 with every transaction.

Reviewer: ```suggestion
Every account carries a nonce property that indicates the number of transactions processed for the given account. Nonces are one-time codes and are incremented by 1 on every transaction.
```

When a new [token transfer transaction](/stacks-blockchain/transactions#stacks-token-transfer) is constructed, the most recent nonce of the account needs to fetched and set.

## Confirmations

The Stacks 2.0 network is anchored onto the bitcoin network. The time to mine a block, in order to confirm transactions, will eventually match the expected "block time" of the bitcoin network: 10 minutes.

The block time is hardcoded and will change throughout the implementation phases of the [testnet](/stacks-blockchain/testnet). The current block time can be obtained through the [`GET /extended/v1/info/network_block_times`](https://blockstack.github.io/stacks-blockchain-api/#operation/get_network_block_times) endpoint:

```bash
curl 'https://stacks-node-api-latest.argon.blockstack.xyz/extended/v1/info/network_block_times'
```

The API will respond with the block time (in seconds):

```js
{
    "testnet": {
        "target_block_time": 120
    },
    "mainnet": {
        "target_block_time": 600
    }
}
```

> -> The finality of the Stacks 2.0 network matches the bitcoin network finality.

Reviewer 2: Can imagine this confusing. Might edit here and add it to the statement at the top of this section: 

"...bitcoin network. This allows transactions on Stacks to inherit the same finality and security of the Bitcoin blockchain." That might be a little too generalized but could make more sense.

Author: much better, thanks!
